### ISSUE TITLE
Could not validate string array items length using JSON schema annotations

### ISSUE THREAD

**Issue by Dancia** [lukeautry/tsoa#962]
First I updated object in swagger (`swagger.json`) schema:
```json
"SomeObject": {
	"arr": {
		"items": {
			"type": "string",
			"minLength": 1,
			"maxLength": 10
		},
		"type": "array"
	}
}
```

Expected routes (`routes.ts`) validation output:
```json
"SomeObject": {
    "dataType": "refObject",
    "properties": {
        "arr": {"dataType":"array","array":{"dataType":"string","validators":{"minLength":{"value":1},"maxLength":{"value":10}}}},
    }
}
```
Problem 1:
But instead I got:
```json
"SomeObject": {
    "dataType": "refObject",
    "properties": {
        "arr": {"dataType":"array","array":{"dataType":"string"}},
    }
}
```

Problem 2:
I could not find a way to represent length validation for array of strings. The object:
```json
"SomeObject": {
	"arr": {
		"items": {
			"type": "string",
		},
		"type": "array"
	}
}
```
was manually updated with `minLength` and `maxLength`. I have tried many different combinations of annotations on `SomeObject`, but not a single one gave the result I expected.
```ts
interface SomeObject {
    /**
     * @type string
     * ... correct annotations?
     */
    arr: string[];
}
```

So basically I see that if `routes.ts` is manually updated, then it gives correct results, but generating `minLength`, `maxLength` validation seems impossible. Am I missing something?

**Comment by WoH** [comment]
> First I updated object in swagger (swagger.json) schema:
[...]
Expected routes (routes.ts) validation output:

That's not how it's supposed to work.

Please fill out the template next time.


```ts
interface SomeObject {
    /**
     * @minLength 5
     */
    arr: string[];
}
```
